compiler/optimizer/VPHandlers.cpp

```diff
@@ -12214,105 +12216,15 @@ TR::Node *constrainArrayStoreChk(OMR::ValuePropagation *vp, TR::Node *node)
          if (base->getFirstChild()->getOpCode().isArrayRef())
             base = base->getFirstChild()->getFirstChild();
          }
-
-      if (base == arrayRef)
-         canBeRemoved = true;
       }
```

Comment:
It looks to me line:12208~12219 are not required any more if `base` is not used after these lines.

Reply:
As I mentioned above, I will be adding a check of `(base == arrayRef)` to determine whether the call to `vp->isArrayStoreCheckNeeded` is needed.

Reply:
Corrected in commit e516f9868d11f2483bd2f917ede6b879a1271006